packages/app-elements/src/ui/resources/ResourceDetails/ResourceDetails.tsx

```diff
@@ -1,41 +1,50 @@
 import { formatDate } from '#helpers/date'
 import { useTokenProvider } from '#providers/TokenProvider'
 import { CopyToClipboard } from '#ui/atoms/CopyToClipboard'
+import { Icon } from '#ui/atoms/Icon'
 import { Section } from '#ui/atoms/Section'
 import { withSkeletonTemplate } from '#ui/atoms/SkeletonTemplate'
 import { Text } from '#ui/atoms/Text'
 import { ListDetailsItem } from '#ui/composite/ListDetailsItem'
-import { type Resource } from '@commercelayer/sdk'
-import isEmpty from 'lodash/isEmpty'
+import { FlexRow } from '#ui/internals/FlexRow'
+import { type ListableResourceType, type Resource } from '@commercelayer/sdk'
+import { useEditDetailsOverlay } from './useEditDetailsOverlay'
 
 export interface ResourceDetailsProps {
   resource: Resource
+  onUpdated: () => Promise<void>
 }
 
 /**
  * This component provides a listed visualization of details attributes of a given resource.
  */
 export const ResourceDetails = withSkeletonTemplate<ResourceDetailsProps>(
-  ({ resource }) => {
-    const { user } = useTokenProvider()
+  ({ resource, onUpdated }) => {
+    const { user, canUser } = useTokenProvider()
+    const { Overlay: EditDetailsOverlay, show } = useEditDetailsOverlay()
+
+    const reference = `${resource?.reference ?? ''}${resource?.reference != null && resource?.reference_origin != null ? ' · ' : ''}${resource?.reference_origin ?? ''}`
```

Comment:
Something like this should work as well

```suggestion
    const reference = [resource?.reference, resource?.reference_origin].filter(Boolean).join(' · ')
```